packages/shared/src/hooks/useBookmarkPost.ts

```diff
@@ -150,10 +165,43 @@ const useBookmarkPost = ({
 
       logEvent(postLogEvent(LogEvent.BookmarkPost, post, logOptions));
 
-      await addBookmark({ id: post.id });
-      displayToast('Post was added to your bookmarks');
+      const { list } = await addBookmark({ id: post.id });
+      if (showPlusSubscription) {
+        displayToast(`Bookmarked! Saved to ${list?.name ?? 'Quick saves'}`, {
+          undoCopy: 'Change folder',
+          onUndo: () => {
+            openModal({
+              type: LazyModal.MoveBookmark,
+              props: {
+                postId: post.id,
+                listId: list?.id,
+                onMoveBookmark: async () => {
+                  logEvent(
+                    postLogEvent(
+                      LogEvent.MoveBookmarkToFolder,
+                      post,
+                      logOptions,
+                    ),
+                  );
```

Comment:
I think we should have this directly from the onSuccess. So that we won't have to always remember whenever we need to implement something similar and miss the analytics. Happened quite many times in the past now.

Reply:
We can do it if we pass the full post object and the logOptions, but at this point it makes sense? @sshanzel

Reply:
Right, there are some differences in the shape of the object. I wouldn't want to block with this, so let's move forward.